app/controllers/api/cars_controller.rb

```diff
@@ -0,0 +1,49 @@
+class Api::CarsController < ApplicationController
+  skip_before_action :set_current_user, only: [:create]
+  def index
+    @reservations = current_user.reservations.includes(:car).where(cars: { removed: false })
```

Comment:
Great using includes its an efficient way to retrieve the necessary data while minimizing the database load.

Reply:
It was very hard, but... Thank you!